presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -361,6 +369,37 @@ private static LongWriteFunction timestampWithTimeZoneWriteFunction()
         };
     }
 
+    private ColumnMapping hstoreColumnMapping()
+    {
+        return ColumnMapping.blockMapping(
+                varcharMapType,
+                varcharMapReadFunction(),
+                (statement, index, block) -> { throw new UnsupportedOperationException(); },
+                DISABLE_PUSHDOWN);
+    }
+
+    private BlockReadFunction varcharMapReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            @SuppressWarnings("unchecked")
+            Map<String, String> map = (Map<String, String>) resultSet.getObject(columnIndex);
```

Comment:
Is this guaranteed to return a map, or do we need to use `resultSet.getObject(columnIndex, Map.class)` ?

Reply:
The latter fails, because `PgResultSet` doesn't support the conversion to `Map`.